adserver/forms.py

```diff
@@ -60,13 +87,231 @@ class Meta:
             "targeting_parameters",
         )
 
-    def clean(self):
-        cpc = self.cleaned_data.get("cpc")
-        cpm = self.cleaned_data.get("cpm")
-        if cpc > 0 and cpm > 0:
-            raise forms.ValidationError(_("A flight cannot have both CPC & CPM"))
 
-        return self.cleaned_data
+class FlightForm(FlightMixin, forms.ModelForm):
+
+    """The form for flights used in a staff interface."""
+
+    include_countries = forms.CharField(
+        max_length=1024,
+        required=False,
+        help_text=_("A comma separated list of country codes"),
+    )
+    exclude_countries = forms.CharField(
+        max_length=1024,
+        required=False,
+        help_text=_("A comma separated list of country codes"),
+    )
+    include_keywords = forms.CharField(
+        max_length=1024,
+        required=False,
+        help_text=_("A comma separated list of keywords"),
+    )
+
+    def __init__(self, *args, **kwargs):
+        """Set the flight form helper and initial data."""
+        super().__init__(*args, **kwargs)
+
+        if self.instance.pk:
+            self.fields["include_countries"].initial = ", ".join(
+                self.instance.included_countries
+            )
+            self.fields["exclude_countries"].initial = ", ".join(
+                self.instance.excluded_countries
+            )
+            self.fields["include_keywords"].initial = ", ".join(
+                self.instance.included_keywords
+            )
+
+        self.helper = FormHelper()
+
+        self.helper.layout = Layout(
+            Fieldset(
+                "",
+                Div(
+                    Div("start_date", css_class="form-group col-lg-6"),
+                    Div("end_date", css_class="form-group col-lg-6"),
+                    css_class="form-row",
+                ),
+                "live",
+                css_class="my-3",
+            ),
+            Fieldset(
+                _("Per impression (CPM) flights"),
+                Div(
+                    Div("cpm", css_class="form-group col-lg-6"),
+                    Div("sold_impressions", css_class="form-group col-lg-6"),
+                    css_class="form-row",
+                ),
+                css_class="my-3",
+            ),
+            Fieldset(
+                _("Per click (CPC) flights"),
+                Div(
+                    Div("cpc", css_class="form-group col-lg-6"),
+                    Div("sold_clicks", css_class="form-group col-lg-6"),
+                    css_class="form-row",
+                ),
+                css_class="my-3",
+            ),
+            Fieldset(
+                _("Flight targeting"),
+                Div(
+                    "include_countries",
+                    Div(
+                        HTML(
```

Comment:
I was trying to do this with crispy forms the other day (render HTML) -- interesting.